src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentBag.cs

```diff
@@ -1052,8 +1052,10 @@ internal int DangerousCount
                 get
                 {
                     Debug.Assert(Monitor.IsEntered(this));
-                    int count = _addTakeCount - _stealCount;
-                    Debug.Assert(count >= 0);
+                    int stealCount = _stealCount;
```

Comment:
should there be `#if DEBUG` around the locals or is the JIT smart enough to drop them as dead code in release?

Reply:
There will probably be an extra mov or two with these locals, but it's not worth having two implementations, one for debug and another for release.  To get here, we would have had to take at least one lock, if not a bunch of locks, and the extra mov or two will be noise.